randomwallpaper@iflow.space/wallpaperController.js

```diff
@@ -313,10 +338,29 @@ var WallpaperController = class {
 					return;
 				}
 
-				historyElement.path = path;
-				historyElement.id = historyId;
+				historyElement.name = String(historyId);
+				historyElement.id = `${historyElement.timestamp}_${historyElement.name}`; // timestamp ensures uniqueness
+
+				// Move file to unique naming
+				let sourceFile = Gio.File.new_for_path(path);
+				let targetFolder = sourceFile.get_parent();
+				let targetFile = targetFolder.get_child(historyElement.id);
+
+				try {
+					if (!sourceFile.move(targetFile, Gio.FileCopyFlags.NONE, null, null)) {
+						this.logger.warning('Failed copying unique image.');
+						return;
+					}
+				} catch (error) {
+					if (error === Gio.IOErrorEnum.EXISTS) {
+						this.logger.warning('Image already exists in location.');
+						return;
+					}
+				}
```

Comment:
I probably missed something, but why is moving the file to a unique name done here instead of in `_fetchFile`?

Reply:
At this point in time `_fetchFile` only returned the ID and the path to the image and I wanted the image name without the unique ID to not save images multiple times.
So the solution was to move up the prepending of the ID, but that also meant I had to rename the file.

However, now in TypeScript the unique ID is already defined in the constructor of the historyEntry and the `_fetchFile` function takes the whole historyEntry so the renaming seems indeed unnecessary.
I'll recheck if I haven't missed anything and will remove the renaming and download directly to the correct name.

Reply:
Ah, thanks for the explanation!